scripts/support/create_ge_images/README.rst

```diff
@@ -0,0 +1,159 @@
+
+The software must be used from a Linux host with the following package installed:
+
+* kvm
+* nova python client
+
+The scripts should be copied to */opt/create_ge_images* to share the installation
+between several users. Also the module *osclients.py*, provided in the parent
+folder, must be copied.
+
+Each user must create the folder *~/create_ge_images* and load the environment
+variables (``OS_USERNAME``, ``OS_PASSWORD``, ``OS_TENANT_NAME``, ``OS_AUTH_URI``) with its
+FiwareLab credential. The script *sec_group* is useful to initialize the
+user account: it creates a pair of security groups (one of them to create
+the image and the other one to test it), generates and registers a key pair and allocates a
+floating IP if it is not already allocated. The scripts expects that the first
+key in *~/.ssh/authorized_keys* is the user's public key.
+
+Creating/testing the image
+**************************
+
+To create an image called *myimage*, the user must create the folder
+*~create_ge_images/myimage* and put inside the scripts *create.sh* and
+*test.sh*. Optionally, the file *data.tgz* can also be added.
+
+It is strongly recommended that both *create.sh* and *test.sh* scripts start with
+``#!/bin/bash -ex``. This way, the script stops when something is wrong and
+a trace is displayed for each command invoked. Anyway, the exit status of the
+script must be zero only when the script runs without error.
+
+After this, one of the scripts *create_template_*.sh*, must be invoked passing
+as parameter the name of the image. For example, to create the image
+*myimage* using the *debian* image, the command to invoke is:
+
+.. code::
+
+  /opt/create_ge_images/create_template_debian.sh myimage
+
+Inside the folder myimage, a pair of logs will be generated: *create.log* is the
+result of invoking the *create.sh* script and *test.log* is the output of
+the *test.sh* launch. Of course, if the execution of the create script fails,
+then only *create.log* is generated.
+
+If all the process ends without errors, the UUID of the new image is printed.
+This image is private and owned by the user, but it is ready to be published
+now.
+
+How the script works
+--------------------
+
+At first, the script launch a VM using the chosen base image, with the generated
+keypair and the security group *sshopen*. The name of the virtual machine is
+the same than the image. Then it assigns the floating IP and wait until a SSH
+connection is ready.
+
+At this moment, the create.sh script is uploaded to the server (the data.tgz
+is upload if it exists, too). Then the script is invoked via SSH.
+
+If the script did not fail, the support account and the script are deleted and
+the server is power down. Then a snapshot is made, downloaded and deleted. Also
+the VM is deleted.
+
+The local copy of the snapshot is *virt-syspred*. Then the shrink process and
+*virt-sparsify* are invoked. Finally, the resulting image is upload as
+*<imagename>_rc* (*rc* means *release candidate*)
```

Comment:
One question, regarding the final name of the image, it includes the version of the GEi?